### DISCUSSION TITLE
Change docker network configuration

### DISCUSSION THREAD

**Discussion by olivierpapin** [telekom-security/tpotce#1402] [Q&A]
Hello everyone,
First of all thank you for this amazing work with T-Pot ! 
I'd like to change the network configuration of the docker containers to use /23 subnets instead of /16 an /28 or /29 subnets for links. I've done it with a daemon.json file in the /etc/docker/ folder but it seems it's not working the same anymore.
Does anyone can help me please ?

**Comment by Ganesh-403** [comment]
@olivierpapin 
T-Pot itself doesn't explicitly define the Docker bridge subnets for most of its internal networks—it relies on Docker's IPAM unless a subnet is specified in the Compose file. So changing `/etc/docker/daemon.json` **can** work, but there are a couple of important caveats.

The biggest one is that **`default-address-pools` only affects newly created Docker networks**. Existing networks keep the subnet they were created with, even after restarting Docker. :contentReference[oaicite:0]{index=0}

So if you've changed `daemon.json`, the general procedure is:

1. Verify Docker has picked up the new configuration:
   ```bash
   docker info
   ```

2. Inspect the current networks:
   ```bash
   docker network ls
   docker network inspect <network_name>
   ```

3. Stop T-Pot:
   ```bash
   sudo systemctl stop tpot
   ```

4. Remove the existing T-Pot Docker networks (or bring the compose stack down so they're recreated):
   ```bash
   docker compose down
   ```
   (or `docker-compose down` on older versions)

5. Start T-Pot again so Docker creates **new** networks using the updated address pools.

If the existing networks aren't removed, they'll continue using their original `/16`, `/28`, or `/29` subnets. :contentReference[oaicite:1]{index=1}

If your goal is specifically to use **/23** bridge networks, your `daemon.json` would need an appropriate `default-address-pools` configuration, for example:

```json
{
  "default-address-pools": [
    {
      "base": "172.20.0.0/16",
      "size": 23
    }
  ]
}
```

Keep in mind that:

- `base` defines the overall address pool.
- `size` defines the subnet prefix allocated to each newly created network. :contentReference[oaicite:2]{index=2}

If, after recreating the networks, T-Pot is **still** creating `/16` or `/28` networks, then it's likely those particular networks are being created with **explicit IPAM settings in the Docker Compose file**, which override Docker's default address pools. In that case, you would need to modify the relevant `networks:` section in the generated `docker-compose.yml` instead of relying on `daemon.json`. Docker gives explicit subnet definitions precedence over the daemon defaults. :contentReference[oaicite:3]{index=3}

Could you share:
- Which **T-Pot version** you're using?
- The output of:
  ```bash
  docker network inspect <one_of_the_tpot_networks>
  ```
- And the relevant `networks:` section from your `docker-compose.yml`?

That will help determine whether the networks are coming from Docker's default address pools or are explicitly defined by the T-Pot compose configuration.

If this solves your problem, feel free to mark it as the accepted answer so others can find it easily.
